Psychtoolbox/PsychCal/MeasureDpi.m

```diff
@@ -16,31 +26,37 @@
 %             both Mac and Windows
 % 11/6/06 dgp Updated from PTB-2 to PTB-3.
 % 01/31/16 mk Fix some wrong assumptions for PTB-3. Get rid of GetChar.
+% 8/13/19 mgg Mofify to use it with a credit card (common object of a
+%             known size).
+% 9/10/19 mgg Optional output: Withdraw the presets from the screen
+%             inc. the screen size, resolution, dot pitch and distance independent dpi.
 
-if nargin>1 || nargout>1
-    error('Usage: dpi=MeasureDpi(screen)');
+if nargin>2 || nargout>2
+    error([datestr(now),':>> Error wrong call to function, use it as: dpi=MeasureDpi(Screen)', newline, 'or [dpi, SP]=MeasureDpi(Screen, CC)']);
 end
-if nargin<1
-    theScreen=0;
+if ~exist(theScreen) || isempty(theScreen)
+    theScreen=max(Screen('Screens')); % Use the second screen, most of the cases.
+    disp([datestr(now),':>> No screen value has been provided so I assume ',...
+        'that either you would like to use the secondary screen, or the main one if you only have one...']);
 end
 AssertOpenGL;
+
 try
-    disp('Please find an object of known width to hold against the display.');
-    disp('E.g. a ruler or an 8.5-inch page.')
-    inches=input('Do you prefer inches (1) or cm (0)? ');
-    if inches
+    if nargin>1 && isequal(CC,1) % Using a credit card
         unitInches=1;
```

Comment:
You don't need to define unitInches, etc. here (and in the else branch) anymore, given that EnglishOrSI() will orverwrite it anyway.